packages/dataviews/src/view-table.js

```diff
@@ -283,62 +301,94 @@ function ViewTable( {
 	isLoading = false,
 	deferredRendering,
 } ) {
+	const headerMenuRefs = useRef( new Map() );
+	const [ nextHeaderMenuToFocus, setNextHeaderMenuToFocus ] = useState();
+
+	useEffect( () => {
+		if ( nextHeaderMenuToFocus ) {
+			nextHeaderMenuToFocus.focus();
+			setNextHeaderMenuToFocus();
```

Comment:
```suggestion
```

Reply:
My thinking was that this would reduce the chances of an unexpected refocus if something else caused the table to be rerendered. We could short-circuit it a bit – it would still cause an extra rerender, but it would be processed immediately without anything expensive happening...

```js
const headerMenuToFocusRef = useRef( null );
const [ nextHeaderMenuToFocus, setNextHeaderMenuToFocus ] = useState();

useEffect( () => {
	if ( headerMenuToFocusRef.current ) {
		headerMenuToFocusRef.current.focus();
		headerMenuToFocusRef.current = null;
	}
} );

if ( nextHeaderMenuToFocus ) {
	headerMenuToFocusRef.current = nextHeaderMenuToFocus;
	setNextHeaderMenuToFocus();
	return;
}
```

Reply:
> My thinking was that this would reduce the chances of an unexpected refocus if something else caused the table to be rerendered.

So, this is an attempt to cover against the situation in which the focus is moved elsewhere _after_ it has been moved to the `headerMenuToFocusRef`, correct? I don't see how setting `headerMenuToFocusRef` to `undefined` would help with re-gaining the focus, as `nextHeaderMenuToFocus.focus();` would never be called in that situation. 

Is there a reproducible situation, or did you run into this while testing? I didn't, so my suggestion would be to remove this extra step. But always happy to reconsider.

Reply:
Ah, so it's the other way around: it's to avoid the header from taking the focus when it has been moved elsewhere. Thanks for the example!

Let's ship as it is and re-consider later.